.github/workflows/build-test.yaml

```diff
@@ -24,32 +24,49 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
+
     - uses: actions/checkout@v2
+
     - name: Install gcc
       run: |
         sudo add-apt-repository ppa:ubuntu-toolchain-r/test
         sudo apt-get update -qq
         sudo apt install gcc-10 g++-10
-    - name: Install python dependencies
-      run: |
-        python -m pip install --upgrade pip setuptools wheel
-        python -m pip install numpy scipy cython nose
+
     - name: Build and install pynbody
       run: |
+        python -m pip install --upgrade pip setuptools wheel
         python -m pip install -e .[tests]
         echo "[ramses]" >> ~/.pynbodyrc
         echo "parallel-read=2" >> ~/.pynbodyrc
+
     - name: Fetch and unpack test data
-      working-directory: nose
       run: |
         wget -q http://star.ucl.ac.uk/~app/testdata.tar.gz
         tar --exclude="._*" -xzvf testdata.tar.gz
+
     - name: Run all tests
-      working-directory: nose
-      run: python -m nose
+      run: python -m pytest --junitxml=pytest.xml
+
     - name: Uninstall posix_ipc
       run: python -m pip uninstall -y posix_ipc
+
     - name: Run ramses tests without shared memory support
-      working-directory: nose
-      run: python -m nose ramses_new_ptcl_format_test.py ramses_test.py
+      run: python -m pytest tests/ramses_new_ptcl_format_test.py tests/ramses_test.py --junitxml=pytest_noshmem.xml
 
+    - name: Upload Unit Test Results
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: Unit Test Results (Python ${{ matrix.python-version }})
+        path: ${{ github.workspace }}/**/pytest*.xml
+
+  event_file:
```

Comment:
As a bonus: it is relatively easy to upload the build results of pytest to Github actions using the following (and the `upload-unit-test-results.yaml` workflow below).